subgraphs/multichain/src/mappings/handlers.ts

```diff
@@ -0,0 +1,834 @@
+    protocol,
+    call.block
+  );
+
+  const tokenAddress = Address.fromString(poolID);
+  const token = getOrCreateToken(protocol, tokenAddress, chainID, call.block);
+
+  updateUsageMetrics(
+    protocol,
+    usageMetricsDaily,
+    usageMetricsHourly,
+    EventType.DEPOSIT,
+    crosschainID,
+    call.block,
+    call.transaction
+  );
+
+  createLiquidityDepositEvent(
+    protocol,
+    token,
+    Bytes.fromHexString(poolID),
+    chainID,
+    call.inputs.to,
+    tokenAddress,
+    call.inputs.amount,
+    call
+  );
+
+  usageMetricsHourly.save();
+  usageMetricsDaily.save();
+  protocol.save();
+}
+
+export function handleDepositVault(call: DepositVaultCall): void {
+  const context = dataSource.context();
+  const poolID = context.getString("poolID");
+  const chainID = BigInt.fromString(context.getString("chainID"));
+  const crosschainID = BigInt.fromString(context.getString("crosschainID"));
+
+  const protocol = getOrCreateProtocol();
+  const usageMetricsDaily = getOrCreateUsageMetricDailySnapshot(
+    protocol,
+    call.block
+  );
+  const usageMetricsHourly = getOrCreateUsageMetricHourlySnapshot(
+    protocol,
+    call.block
+  );
+
+  const tokenAddress = Address.fromString(poolID);
+  const token = getOrCreateToken(protocol, tokenAddress, chainID, call.block);
+
+  updateUsageMetrics(
+    protocol,
+    usageMetricsDaily,
+    usageMetricsHourly,
+    EventType.DEPOSIT,
+    crosschainID,
+    call.block,
+    call.transaction
+  );
+
+  createLiquidityDepositEvent(
+    protocol,
+    token,
+    Bytes.fromHexString(poolID),
+    chainID,
+    call.inputs.to,
+    tokenAddress,
+    call.inputs.amount,
+    call
+  );
+
+  usageMetricsHourly.save();
+  usageMetricsDaily.save();
+  protocol.save();
+}
+
+export function handleWithdraw(call: WithdrawCall): void {
+  const context = dataSource.context();
+  const poolID = context.getString("poolID");
+  const chainID = BigInt.fromString(context.getString("chainID"));
+  const crosschainID = BigInt.fromString(context.getString("crosschainID"));
+
+  const protocol = getOrCreateProtocol();
+  const usageMetricsDaily = getOrCreateUsageMetricDailySnapshot(
+    protocol,
+    call.block
+  );
+  const usageMetricsHourly = getOrCreateUsageMetricHourlySnapshot(
+    protocol,
+    call.block
+  );
+
+  const tokenAddress = Address.fromString(poolID);
+  const token = getOrCreateToken(protocol, tokenAddress, chainID, call.block);
+
+  updateUsageMetrics(
+    protocol,
+    usageMetricsDaily,
+    usageMetricsHourly,
+    EventType.WITHDRAW,
+    crosschainID,
+    call.block,
+    call.transaction
+  );
+
+  createLiquidityWithdrawEvent(
+    protocol,
+    token,
+    Bytes.fromHexString(poolID),
+    chainID,
+    tokenAddress,
+    call.inputs.to,
+    call.inputs.amount,
+    call
+  );
+
+  usageMetricsHourly.save();
+  usageMetricsDaily.save();
+  protocol.save();
+}
+
+export function handleWithdrawVault(call: WithdrawVaultCall): void {
```

Comment:
What's the difference between Deposit & DepositVault and Withdraw&WithdrawVault?